version.py

```diff
@@ -40,30 +40,49 @@
 
 
 def call_git_describe(prefix='', cwd='.'):
-    cmd = 'git describe --tags --match %s[0-9]*' % prefix
-    try:
-        version = subprocess.run(
-            cmd.split(),
-            stdout=subprocess.PIPE,
-            check=True,
-        ).stdout.decode()
-        version = version.strip()[len(prefix):]
-        if '-' in version:
-            version = '{}+dev{}.{}'.format(*version.split('-'))
-        return version
-    except Exception:
-        return None
+    version_cmd = 'git describe --tags --match %s[0-9]*' % prefix
+    last_tag_cmd = 'git describe --tags --match %s[0-9]* --abbrev=0' % prefix
+    version = subprocess.run(
+        version_cmd.split(),
+        stdout=subprocess.PIPE,
+        check=True,
+        shell=False,
+    ).stdout.decode()
+
+    last_tag = subprocess.run(
+        last_tag_cmd.split(),
+        stdout=subprocess.PIPE,
+        check=True,
+        shell=False,
+    ).stdout.decode()
+
+    assert ' ' not in prefix
+    version = version.strip()[len(prefix):]
+    last_tag = last_tag.strip()[len(prefix):]
+    if version != last_tag:
+        """
+        Helper to generate version compatible with semver
+        Compares the last tag version name with the version name
+        where the repository is currently located
+        If it is not currently on tag adds '+dev' beetwen last tag name
+        and a number of commits behind it separated by hyphens with short commit hash
+        """
+        version = f"{last_tag}+dev{version[len(last_tag) + 1:].replace('-', '.')}"
+    return version
 
 
-def get_version(prefix='', cwd='.'):
+def get_version(prefix='', cwd='.', generate_version=True):
     path = pathlib.Path(cwd) / VERSION_FILE
     try:
         with path.open("r") as f:
             release_version = f.read()
-    except Exception:
+    except FileNotFoundError:
         release_version = None
 
-    version = call_git_describe(prefix, cwd)
+    if generate_version:
+        version = call_git_describe(prefix, cwd)
+    else:
+        return release_version
```

Comment:
When `generate_version` is `False` and `RELEASE-VERSION` does not exist the script should fail.

You can do this by raising an exception there or later, inside the `__main__` block.

Reply:
Do you want me to do specific exception or just `Exception`?

Reply:
I would personally always use a custom exception class but since the code is already using `ValueError` raising it is fine too.

Reply:
So it is done